textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,311 @@
+options {
+    STATIC = false;
+    UNICODE_INPUT = true;
+}
+
+/*
+Compiler Command for JavaCC:
+~:/home/virtualbox
+root: ~/Desktop/textdb/textdb/textdb-textql/src/main/
+JavaCC files: root/javacc
+Java Generated Files: root/java/edu/uci/ics/textdb/textql/languageparser
+
+javacc \
+-OUTPUT_DIRECTORY:/home/virtualbox/Desktop/textdb/textdb/textdb-textql/src/main/java/edu/uci/ics/textdb/textql/languageparser \
+/home/virtualbox/Desktop/textdb/textdb/textdb-textql/src/main/javacc/TextQLParser.jj
+
+*/
+
+PARSER_BEGIN(TextQLParser)
+
+package edu.uci.ics.textdb.textql.languageparser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class TextQLParser{    
+    
+    private int _lid = 0;
+    private String getNewUniqueId(){
+        return "__lid" + Integer.toString(_lid++, 36);
+    }
+}
+
+PARSER_END(TextQLParser)
+
+/*
+ *	Skip Tokens
+ */
+SKIP : { " " | "\n" | "\r" | "\t" | "\f" }
+
+/*
+ *	Basic Tokens
+ */
+TOKEN : { < CREATE : "CREATE" > }
+TOKEN : { < DICTIONARY : "DICTIONARY" > }
+TOKEN : { < VIEW : "VIEW" > }
+TOKEN : { < SELECT : "SELECT" > }
+TOKEN : { < EXTRACT : "EXTRACT" > }
+TOKEN : { < FROM : "FROM" > }
+TOKEN : { < AS : "AS" > }
+TOKEN : { < LIMIT : "LIMIT" > }
+TOKEN : { < OFFSET : "OFFSET" > }
+TOKEN : { < USING : "USING" > }
+TOKEN : { < ON : "ON" > }
+TOKEN : { < JOIN : "JOIN" > }
+TOKEN : { < WHERE : "WHERE" > }
+TOKEN : { < OUTPUT : "OUTPUT" > }
+TOKEN : { < INSERT : "INSERT" > }
+TOKEN : { < INTO : "INTO" > }
+TOKEN : { < VALUES : "VALUES" > }
+TOKEN : { < TO : "TO" > }
+TOKEN : { < ID : "ID" > }
+
+TOKEN : { <LPAREN: "(" > }
+TOKEN : { < RPAREN: ")" > }
+TOKEN : { < LBRACE: "{" > }
+TOKEN : { < RBRACE: "}" > }
+TOKEN : { < LBRACKET: "[" > }
+TOKEN : { < RBRACKET: "]" > }
+TOKEN : { < SEMICOLON: ";" > }
+TOKEN : { < COMMA: "," > }
+TOKEN : { < DOT: "." > }
+TOKEN : { < STAR : "*" > }
+
+TOKEN : { < ASSIGN: "="> }
+TOKEN : { < LT: "<" > }
+TOKEN : { < GT: ">" > }
+TOKEN : { < EQ: "==" > }
+TOKEN : { < LE: "<=" > }
+TOKEN : { < GE: ">=" > }
+TOKEN : { < NE: "!=" > }
+
+/*
+ *	Literal Tokens
+ */
+TOKEN : { < NUMBER_LITERAL : ("-")? ((["0"-"9"])+ ("." (["0"-"9"])*)? | "." (["0"-"9"])+) > }
+TOKEN : { < STRING_LITERAL : "\"" ( "\\" ~[] | ~["\""] )* "\"" > }
+TOKEN : { < REGEX_LITERAL : "/" ( "\\" ~[] | ~["/"] )* "/" > }
+TOKEN : { < IDENTIFIER : ["a"-"z"] (["a"-"z","0"-"9"])* > }
```

Comment:
Add a comment to say that those names have to be lower case.  Add a TODO to get rid of this restriction.  Add an example.